### ISSUE TITLE
Toggling hotpost on/off doesn't necesarilly reflect in the indicator or system settings

### ISSUE THREAD

**Issue by zubozrout** [ubports/ubports-touch#281]
- Device: MX4
- Channel: devel (any, even Canonical images)
- Build: r38

## Steps to reproduce

1. Toggle Hotspot on in the indicator area
2. Go to System Settings → Hotspot and see it turned off (you can now toggle Hotspot on again)
3. So do it - toggle the Hotspot on in System Settings, then off again and you can still see it turned on in the indicator area (since the Hotspot icon doesn't show up for me almost ever you can only tell by the switch turned on)

However, turning the Hotspot on in the System Settings app seems to be reflected in the indicator for me.

## Expected behavoir
Reflect changes to the Hotspot state on both places: System Settings and Indicator area - ideally update those each time they are accessed or immediately if Hotspot is activated and indicator or System Settings visible (this does seem to work for the indicator if Hotspot is turned on via System Settings). Or at the very least check the state on accessing the Hotspot entry in System Settings or sliding down the network indicator pane.

## Possibly relevant issue
This may be relevant to this bug as it seems as though there may be some updating error in the indicator itself not reporting correct Hotspot state to the relevant places: https://github.com/ubports/ubports-touch/issues/41

Thank you very much for looking into this.

**Comment by userj** [comment]
Same experience  on oneplus one, bacon

**Comment by NeoTheThird** [comment]
Yeah, i can confirm this behavior for the Oneplus One, but in my experience it worked a lot better on the FP2.

**Comment by ruedigerkupper** [comment]
Confirmed for MX4

**Comment by NeoTheThird** [comment]
Don't mind me, brain fart.

**Comment by zubozrout** [comment]
I seem to no longer experience this behaviour on MX4. Meaning the indicators reflect the change when I actually succeed switching the hotspot on. Albeit in the System Settings app leaving the hotspot page and accessing it again is required to see the updated state. But I guess that's more of an inconvenience than an actual bug.

**Comment by UniversalSuperBox** [comment]
This issue appears closely related to #526. If the device is connected to WiFi at the time of turning on the hotspot, this is more likely to occur. When the error occurs, indicator-network says:

```
Debug: Updating hotspot connection ((null):0, (null))
Debug: Activating hotspot on device "/org/freedesktop/NetworkManager/Devices/5" ((null):0, (null))
Debug: Waiting for hotspot to connect ((null):0, (null))
(repeats 20 times)
```
